src/app/components/add-card/add-card.component.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  Component,
+  OnInit,
+  ElementRef,
+  ViewChild,
+  Input,
+  Output,
+  EventEmitter,
+} from '@angular/core';
+import { formatDate } from '@angular/common';
+import * as DOMPurify from 'dompurify';
+
+import { UtilityService } from 'src/app/utility.service';
+import { TaskService } from 'src/app/task.service';
+import { ConstantsService } from 'src/app/constants.service';
+
+@Component({
+  selector: 'app-add-card',
+  templateUrl: './add-card.component.html',
+  styleUrls: ['./add-card.component.css'],
+})
+export class AddCardComponent implements OnInit {
+  constructor(
+    public utilityService: UtilityService,
+    public taskService: TaskService,
+    public constantsService: ConstantsService
```

Comment:
Why public instead of private, can you share your thought?

Reply:
By adding public to the constructor parameters, TypeScript will automatically create and initialize the corresponding class properties, allowing us to access them within the class. an easier way to create and initialize class properties. following in an alternative way:
```
 utilityService: UtilityService;
  taskService: TaskService;
  constantsService: ConstantsService;
  constructor(
    utilityService: UtilityService,
    taskService: TaskService,
    constantsService: ConstantsService
  ) {
    this.utilityService = utilityService;
    this.taskService = taskService;
    this.constantsService = constantsService;
  }
```
but if it is concerned with private access specifier, Using private access specifier is not allowing other files of AddCardComponent to access the properties.